src/rgw/rgw_op.cc

```diff
@@ -4285,7 +4285,7 @@ int RGWDeleteObj::verify_permission()
     if (r == Effect::Allow)
       return true;
     else if (r == Effect::Deny)
-      return false;
+      return -EACCES;
```

Comment:
Good catch, could you also return 0 instead of true; and since you're here anyway can you also cleanup the usr_policy_res above (returning -EACCESS instead of false)

Reply:
Also @adamemerson / @pritha-srivastava  do we need to both Bucket Policy and user policy to return `Allow` to succeed or is it an OR here ?

Reply:
Yes its an OR here ( basically the bucket policy returns an Effect::Pass at this point)

Reply:
@theanalyst Done~

Reply:
@pritha-srivastava 
Is there a scenario that we set both bucket policy and user policy to an object? Is there any priority judging in this scenario?

Reply:
I think at least more of the overlap should be better defined.
@ZVampirEM77 can we get a unit test for this?

Reply:
@ZVampirEM77 : In AWS User Policies are mostly used for services other than S3, but can be used for s3 also, and bucket policies are for s3 only. 
In RGW, a user policy can be defined for controlling a createBucket and ListAllBuckets operations(before bucket creation), and then fine grained control over other ops for a bucket can be via bucket policies.
There is no priority of one over another. The result is an intersection of all policies set. Check this page: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic.html

Reply:
We examine all the relevant policies and so long as at least ONE returns allow and NONE return Deny we allow the operation.

Reply:
> @ZVampirEM77 can we get a unit test for this?
@mattbenjamin Yeah, but where should we add the unit test? I think currently we have no any unit test for DeleteObj with policy,  what's more, the unit tests for PutObj with policy and GetObj with policy is also lacking.

Reply:
@ZVampirEM77 You can find s3tests related to Bucket Policies here: https://github.com/ceph/s3-tests/blob/master/s3tests/functional/test_s3.py#L8817